tests/TestableIO.System.IO.Abstractions.TestingHelpers.Tests/MockFileStreamTests.cs

```diff
@@ -390,4 +390,26 @@ async Task Act() =>
             await source.CopyToAsync(destination);
         await That(Act).Throws<NotSupportedException>();
     }
+
+    [Test]
+    public async Task MockFileStream_WhenExclusiveStreamOpen_ShouldThrowIOException()
+    {
+        var fileSystem = new MockFileSystem();
+        fileSystem.File.WriteAllText("foo.txt", "");
+        using (new MockFileStream(fileSystem, "foo.txt", FileMode.Open, FileAccess.Read, FileShare.None))
+        {
+            await That(() => new MockFileStream(fileSystem, "foo.txt", FileMode.Open, FileAccess.Read)).Throws<IOException>();
+        }
+    }
+
```

Comment:
I think you could combine these two tests into one and verify that after you dispose of the stream it no longer throws...